### PR TITLE
Change Pagination Method

### PR DESCRIPTION
**Pull request by WillKoehrsen**
- Updates dataset fetching to account for new pagination method in the interval server
- Not a breaking change because the interval server maintains backwards compatibility with the offset based pagination

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -37,7 +37,7 @@ gridstatus = "^0.20.0"
 pytest = "7.1.2"
 pytest-xdist = "3.0.2"
 pytest-rerunfailures = "10.3"
-ruff = "0.0.202"
+ruff = "0.0.291"
```
Comment (WillKoehrsen) [inline pyproject.toml:40]: Ruff >=0.0.291 is required to use the [`Ruff` VSCode extension](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff).

File: gridstatusio/tests/test_api.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
 
 client = gs.GridStatusClient(
     api_key=os.getenv("GRIDSTATUS_API_KEY_TEST"),
+    host=os.getenv("GRIDSTATUS_HOST_TEST"),
```
Comment (WillKoehrsen) [inline gridstatusio/tests/test_api.py:11]: Useful for local testing.

File: gridstatusio/gs_client.py [hunk 3]
```diff
@@ -276,6 +278,7 @@ def get_dataset(
         has_next_page = True
         dfs = []
         total_time = 0
+        total_rows = 0
```
Comment (WillKoehrsen) [inline gridstatusio/gs_client.py:281]: To improve usability, if `limit` is provided, then we could create a progress bar.

File: gridstatusio/gs_client.py [hunk 4]
```diff
@@ -201,6 +204,7 @@ def get_dataset(
         page_size=None,
         tz=None,
         verbose=True,
+        use_cursor_pagination=True,
```
Comment (kmax12) [inline gridstatusio/gs_client.py:207]: let's add this to the docstring. 

also, do we want to do a release where we dont change the defaults, just we can test ourselves that everything is working well? 

one test i want to do: for every dataset,

```python
a = client.get_dataset(dataset_id, limit=100)
b = client.get_dataset(dataset_id, limit=100, page_size=10)
assert a.equals(b)
```

**Review by kmax12 — APPROVED** [review]

File: gridstatusio/gs_client.py (hunk 4 above)
Comment (WillKoehrsen) [reply gridstatusio/gs_client.py:207]: Yeah, I think we can set the default to `False`.

File: gridstatusio/gs_client.py [hunk 5]
```diff
@@ -88,20 +88,23 @@ def get(self, url, params=None, verbose=False):
...
             data = response.json()
             df = pd.DataFrame(data["data"][1:], columns=data["data"][0])
+            meta = data["meta"]
         elif self.request_format == "csv":
             df = pd.read_csv(io.StringIO(response.text), low_memory=False)
 
         has_next_page = response.headers["x-has-next-page"] == "true"
 
-        return df, has_next_page
+        return df, has_next_page, meta
```
[the remark this answers is not in the document]
Comment (WillKoehrsen) [reply gridstatusio/gs_client.py:107]: Yes, removed the `has_next_page` variable.

**Comment by WillKoehrsen** [comment]
One of the tests is failing, so I'll hold off on merging this. I think the issue is on the server side.

**Pull request merged**